### ISSUE TITLE
Unhandled exception in 3.7.0 after setstatus $EXPID -s -t COMPLETED -fl $JOB

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#174]
In GitLab by @jordicuadrado on Aug 5, 2016, 11:54

Hi AS team,

I was testing the 3.7.0 version for expid a001: `/home/Earth/jcuadrad/autosubmit/a001/`

A local job failed (`/home/Earth/jcuadrad/autosubmit/a001/tmp/LOG_a001/a001_LOCAL_SEND_STATIC.cmd line 97`)
producing the log at: `/home/Earth/jcuadrad/autosubmit/a001/tmp/LOG_a001/a001_LOCAL_SEND_STATIC.cmd.20160805095724.err`
however, the job never left the `UNKNOWN` status.

I manually fixed the softlink and executed: `autosubmit setstatus a001 -s -t COMPLETED -fl a001_LOCAL_SEND_STATIC`

Then, trying to re-run the experiment the following exception was raised:

```bash
jcuadrad@ bscearth327:~/autosubmit/a001> autosubmit run a001
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
autosubmit_a001.conf OK
platforms_a001.conf OK
jobs_a001.conf OK
expdef_a001.conf  OK
Configuration files OK

Starting job submission...
Creating jobs...
Loading JobList
[CRITICAL] Unhandled exception on Autosubmit: iteration over non-sequence
Traceback (most recent call last):
  File "/shared/earth/software/autosubmit/3.7.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.0-py2.7.egg/autosubmit/autosubmit.py", line 313, in parse_args
    return Autosubmit.run_experiment(args.expid)
  File "/shared/earth/software/autosubmit/3.7.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.0-py2.7.egg/autosubmit/autosubmit.py", line 592, in run_experiment
    job_list = Autosubmit.load_job_list(expid, as_conf)
  File "/shared/earth/software/autosubmit/3.7.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.0-py2.7.egg/autosubmit/autosubmit.py", line 2197, in load_job_list
    date_format, as_conf.get_retrials(), as_conf.get_default_job_type(), False)
  File "/shared/earth/software/autosubmit/3.7.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.0-py2.7.egg/autosubmit/job/job_list.py", line 112, in generate
    jobs_data = {str(row[0]): row for row in self.load()}
TypeError: iteration over non-sequence
```

All files are left untouched under /home/Earth/jcuadrad/autosubmit/a001/ if you need them for further analysis.

Maybe is an stupid thing to report, but hope this helps for your testing period of AS 3.7.0,

Thanks!

**Comment by kinow** [comment]
In GitLab by @dmanubens on Aug 5, 2016, 17:13

There is a bug in set_status function. Develop branch has been updated. Meanwhile you could use recovery as an alternative.
Thanks for reporting!

**Comment by kinow** [comment]
In GitLab by @dmanubens on Aug 12, 2016, 17:55

Status changed to closed by commit 5d1278c6ddd13ee9cc3aefcfe80bc50e7de9c8f5
